airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py

```diff
@@ -115,9 +116,18 @@ def _wrap_refresh_token_exception(self, exception: requests.exceptions.RequestEx
     def _get_refresh_access_token_response(self) -> Any:
         try:
             response = requests.request(method="POST", url=self.get_token_refresh_endpoint(), data=self.build_refresh_request_body())
-            self._log_response(response)
-            response.raise_for_status()
-            return response.json()
+            if response.ok:
+                response_json = response.json()
+                # Add the access token to the list of secrets so it is replaced before logging the response
+                # An argument could be made to remove the prevous access key from the list of secrets, but unmasking values seems like a security incident waiting to happen...
+                access_key = response_json[self.get_access_token_name()]
```

Comment:
I'm a little bit confused why we need this extra logic here - `refresh_token` is already being masked out in the refresh access token response, so I figured we had just forgotten to also specify the access token in the same place that we say `refresh_token` should be obfuscated. Why do we need extra logic for access token?

Reply:
there won't be an access token if the request is not successful